process/txcost/transactionCostProcessor.go

```diff
@@ -201,5 +211,53 @@ func (tcp *transactionCostProcessor) processScResult(
 		return nil, err
 	}
 
+	tcp.hasExecutedSCR = true
+
 	return res, nil
 }
+
+func (tcp *transactionCostProcessor) extractCorrectResponse(txSender string, currentRes *data.TxCostResponseData) *data.TxCostResponseData {
+	if len(tcp.responses) == 1 {
+		return currentRes
+	}
+
+	for _, res := range tcp.responses {
+		if hasToMuchGasProvidedTopic(&res.Data) || hasSCRWithRefundForSender(txSender, &res.Data) {
+			return &res.Data
+		}
+	}
+
+	return currentRes
+}
+
+const tooMuchGasProvidedMessage = "@too much gas provided"
+
+func hasToMuchGasProvidedTopic(res *data.TxCostResponseData) bool {
```

Comment:
could have been `doEventsContainTopic(res *data.TxCostResponseData, topic string) bool` so you could reuse it for other topics as well

Reply:
refactored